web/assets/js/components/gameoptions.ts

```diff
@@ -1,237 +1,283 @@
+
+        this.DOM.paneConnection = this.DOM.el.querySelector("#js-connection");
+        this.DOM.paneCategory = this.DOM.el.querySelector("#js-category");
+        this.DOM.paneTimer = this.DOM.el.querySelector("#js-timer");
+        this.DOM.paneLoader = this.DOM.el.querySelector("#js-paneloader");
+        this.DOM.categoryBtns = [...this.DOM.el.querySelectorAll(".js-categoryUpdate")];
+        this.DOM.categorySwitch = this.DOM.el.querySelector("#js-categorySwitch");
+        this.DOM.timerBtns = [...this.DOM.el.querySelectorAll(".js-timerUpdate")];
+        this.DOM.timerDisplay = this.DOM.el.querySelector("#js-timerdisplay");
+        this.DOM.timerIncrement = this.DOM.el.querySelector("#js-timerincrement");
+        this.DOM.ctrl0 = this.DOM.el.querySelector("#js-gameoptions-ctr0");
+        this.DOM.ctrl1 = this.DOM.el.querySelector("#js-gameoptions-ctr1");
+
+        //controls events
+        //TODO: keybord press events
+        this.events.clickOnCtrl0 = this.on({
+            e: "click",
+            target: this.DOM.ctrl0,
+            cb: this._clickOnCtrl0.bind(this),
+        });
+        this.events.clickOnCtrl1 = this.on({
+            e: "click",
+            target: this.DOM.ctrl1,
+            cb: this._clickOnCtrl1.bind(this),
+        });
+
+        //category events
+        this.DOM.categoryBtns.forEach((categoryBtn: Element, i: number) => {
+            this.events["clickOnCategory" + i] = this.on({
+                e: "change",
+                target: categoryBtn,
+                cb: this._updateCategory.bind(this),
+            });
+        });
+
+        //timer events
+        this.DOM.timerBtns.forEach((timerBtn: Element, i: number) => {
+            this.events["clickOnTimer" + i] = this.on({
+                e: "click",
+                target: timerBtn,
+                cb: this._updateTimer.bind(this),
+            });
         });
-      }
+
+        //tl
+        this.switchAnimation1 = gsap
+            .timeline({paused: true})
+            .to(this.DOM.paneConnection, {autoAlpha: 0, display: "none", duration: 0.6})
+            .to(this.DOM.paneCategory, {autoAlpha: 1, display: "flex", duration: 0.6})
+            .to(this.DOM.paneTimer, {autoAlpha: 1, display: "flex", duration: 0.6}, "<");
+
+        this.switchAnimation2 = gsap
+            .timeline({paused: true})
+            .to(this.DOM.paneCategory, {autoAlpha: 0, display: "none", duration: 0.6})
+            .to(this.DOM.paneTimer, {autoAlpha: 0, display: "none", duration: 0.6}, "<")
+            .to(this.DOM.ctrl1, {autoAlpha: 0, duration: 0.6}, "<")
+            .to(this.DOM.paneLoader, {autoAlpha: 1, display: "flex", duration: 0.6});
+
+        //actions
+        this.DOM.timerDisplay.innerHTML = this.options.timer[0];
+        this.DOM.timerIncrement.innerHTML = this.options.timer[1];
+
+        //checkstep
+        Actions.getInstance().then((actions) => {
+            // TODO @Alexis please check that the semantics are right here
```

Comment:
The semantics? Here, when the options component (the options DOM panel) is init, it checks if the token has already been registered, if so, it passes the first screen (the "provide token" first option screen) by emulating a click on the "next" button. So no step 0 but step 1 instead. Your code, seems good.

Reply:
Got it, thank you 🙏